index.js

```diff
@@ -93,6 +93,27 @@ function checkTopicQueryParameter(req, res, next) {
     next();
 }
 
+
+function checkQoSParameter(req,res,next){
+
+  if(!req.body.qos){
+    req.body.qos=0
+  }
+  if( 0 > parseInt(req.body.qos, 10) || parseInt(req.body.qos, 10)  >2){
```

Comment:
no need to parse and check req.body.qos if we know we already set it (see lines 99-101).

Do something like:
```javascript
if (req.body.qos){
  req.body.qos = 0;
  next();
}
else {
  let qos = parseInt(req.body.qos, 10);

  if (qos < 0 || qos > 2){
    res.status(400).send("QoS range is not in range 0-2");
    // do not call next()
  }
  else {
    next();
  }
}

```